specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json

```diff
@@ -0,0 +1,9155 @@
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "schema": {
+              "$ref": "#/definitions/EnableInteractiveQueryForIntegrationRuntimeRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/IActionResult"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "IntegrationRuntimes_EnableInteractiveQuery": {
+            "$ref": "./examples/IntegrationRuntimes_EnableInteractiveQuery.json"
+          }
+        }
+      }
+    },
+    "/managedvirtualnetworks/{managedVNetName}": {
+      "get": {
+        "tags": [
+          "ManagedVirtualNetworks"
+        ],
+        "description": "Gets a ManagedVirtualNetwork",
+        "operationId": "ManagedVirtualNetworks_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "managedVNetName",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "default"
+            ],
+            "x-ms-enum": {
+              "name": "ManagedVirtualNetworkName",
+              "modelAsString": true
+            }
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ManagedVirtualNetwork"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ManagedVirtualNetworks_Get": {
+            "$ref": "./examples/ManagedVirtualNetworks_Get.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ManagedVirtualNetworks"
+        ],
+        "description": "Creates or Updates a ManagedVirtualNetwork",
+        "operationId": "ManagedVirtualNetworks_CreateOrUpdate",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "managedVNetName",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "default"
+            ],
+            "x-ms-enum": {
+              "name": "ManagedVirtualNetworkName",
+              "modelAsString": true
+            }
+          },
```

Comment:
Why is this defined as an enum with only one value? Do you expect to allow additional values later? Will these be fixed values or will the user be able to assign their own values? I am concerned that we many not be able to evolve this pattern compatibly.

Reply:
Right now UI doesn't have a way to create a ManagedVirtualNetwork. It implicitly calls service with the name "default" while creating ManagedIntegrationRuntime, I think. @JiaxingSong, please provide more context here if I'm missing anything. 


Since we are not letting users create a ManagedVirtualNetwork with any other name besides "default", I had to make this an enum. I'm not sure if there are plans to let users create multiple VirtualNetworks with names of their own choice.

Reply:
Today we allow only 'default' as virtual network name, this is because managed vnet runtime feature allows only one vnet for each purview account. In the future we probably will allow user to create more vnet with different names, so @pvadali-az using a enum here will block the flexibility in the coming feature evolvements, I'd prefer to allow arbitrary string as the name and throw 400 from service side if the input name is other than 'default'.   In this way, we could easily allow more values from service implementation instead of release new SDK version